examples/cat_facts/cli/src/core.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{anyhow, Result};
+use async_std::{
+    fs::{File, OpenOptions},
+    io::{ReadExt, WriteExt},
+    task::spawn,
+};
+use chrono::{DateTime, Utc};
+use crossbeam_channel::Sender;
+use std::{sync::Arc, time::SystemTime};
+use tracing::debug;
+
+use shared::{
+    key_value::{KeyValueOperation, KeyValueOutput},
+    platform::PlatformResponse,
+    time::TimeResponse,
+    CatFactCapabilities, CatFacts, Effect, Event,
+};
+
+use crate::http;
+
+pub type Core = Arc<shared::Core<Effect, CatFacts>>;
```

Comment:
I tried that, but when we spawn, because of the recursion, we need the lifetime of the instance to outlive 'static. So having them as static functions (which we pass a ref to core into) makes this much easier. And static functions on a mod called `core` is not really different to static methods on a struct called `Core`.

Reply:
Ah ok, let's leave it then.